packages/babel-preset-typescript/src/plugin-rewrite-ts-imports.ts

```diff
@@ -1,25 +1,50 @@
 import { declare } from "@babel/helper-plugin-utils";
-import type { types as t, NodePath } from "@babel/core";
+import type { types as t, NodePath, PluginPass } from "@babel/core";
 
 export default declare(function ({ types: t, template }) {
   function maybeReplace(
-    source: t.ArgumentPlaceholder | t.SpreadElement | t.Expression,
+    source: t.ArgumentPlaceholder | t.Expression,
     path: NodePath,
+    state: PluginPass,
   ) {
     if (!source) return;
+    // todo: if we want to support `preserveJsx`, we can register a global flag via file.set from transform-react-jsx, and read it here.
+    const preserveJsx = false;
     if (t.isStringLiteral(source)) {
-      if (/[\\/]/.test(source.value)) {
-        source.value = source.value
-          .replace(/(\.[mc]?)ts$/, "$1js")
-          .replace(/\.tsx$/, ".js");
+      if (/^\.\.?\//.test(source.value)) {
```

Comment:
```suggestion
      if (source.value.startsWith(".")) {
```

Reply:
This regex tests about `./` and `../`, so it can not be replaced by the suggestion.